vllm/v1/worker/gpu_model_runner.py

```diff
@@ -382,7 +382,8 @@ def execute_model(
                 # Rewind the generator state as if the token was not sampled.
                 generator = self.input_batch.generators.get(i)
                 if generator is not None:
-                    generator.set_offset(generator.get_offset() - 1)
+                    # This relies on cuda-specific torch-internal impl details
+                    generator.set_offset(generator.get_offset() - 4)
```

Comment:
@WoosukKwon we may need to figure out something different out here. Can we incorporate this condition into the sampling metadata so that the sampler doesn't sample the tokens that we don't need?